pkgs/applications/misc/protonup-qt/default.nix

```diff
@@ -0,0 +1,32 @@
+{ appimageTools, fetchurl, lib }:
+let
+  pname = "protonup-qt-bin";
+  name = "protonup-qt";
+  version = "2.7.4";
+  src = fetchurl {
+    url = "https://github.com/DavidoTek/ProtonUp-Qt/releases/download/v${version}/ProtonUp-Qt-${version}-x86_64.AppImage";
+    sha256 = "yKc+KOQfqciqULnChVLf6y9npoSYM6Fmu7mYGEKmpkA=";
+  };
+  appimageContents = appimageTools.extractType2 { inherit pname version src; };
+in
+appimageTools.wrapType2 rec {
+  inherit name pname version src;
+
+  extraInstallCommands = ''
+    mkdir -p $out/share/{applications,pixmaps}
+    cp ${appimageContents}/net.davidotek.pupgui2.desktop $out/share/applications/${pname}.desktop
+    cp ${appimageContents}/net.davidotek.pupgui2.png $out/share/pixmaps/${pname}.png
+    substituteInPlace $out/share/applications/${pname}.desktop \
+      --replace 'Exec=net.davidotek.pupgui2' 'Exec=${name}' \
+      --replace 'Icon=net.davidotek.pupgui2' 'Icon=${pname}'
```

Comment:
It is set to `Exec=${name}` as `name` is the name given to the resulting binary.

Is there a reason to use `Icon=protonup-qt-bin` instead of `Icon=${pname}` when the icon will always be named using `pname`?

Reply:
I just tried to guess what the correct values would be. If they are wrong then please use the correct ones.

I am not a big fan of substituting pname where possible but if they name of the file depends on pname then feel free to use it.